components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UsersResource.java

```diff
@@ -0,0 +1,150 @@
+                    break;
+                }
+            }
+        } catch (UserStoreException e) {
+            response = Utils.createJsonError("Error initializing the user store. Please try again later", e,
+                                             axis2MessageContext, INTERNAL_SERVER_ERROR);
+        } catch (IOException e) {
+            response = Utils.createJsonError("Error processing the request", e, axis2MessageContext, BAD_REQUEST);
+        }
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        Utils.setJsonPayLoad(axis2MessageContext, response);
+        return true;
+    }
+
+    JSONObject handleGet(MessageContext messageContext) throws UserStoreException {
+        LOG.debug("Handling GET");
+
+        String searchPattern = Utils.getQueryParameter(messageContext, PATTERN);
+        if (Objects.isNull(searchPattern)) {
+            searchPattern = "*";
+        }
+
+        List<String> patternUsersList = Arrays.asList(getUserStore().listUsers(searchPattern, -1));
+
+        String roleFilter = Utils.getQueryParameter(messageContext, ROLE);
+        JSONObject jsonBody;
+        List<String> users;
+        if (Objects.isNull(roleFilter)) {
+            users = patternUsersList;
+        } else {
+            List<String> roleUserList = Arrays.asList(getUserStore().getUserListOfRole(roleFilter));
+            users = patternUsersList.stream().filter(roleUserList::contains).collect(Collectors.toList());
+        }
+        jsonBody = Utils.createJSONList(users.size());
+        for (String user : users) {
+            JSONObject userObject = new JSONObject();
+            userObject.put(USER_ID, user);
+            jsonBody.getJSONArray(LIST).put(userObject);
+
+        }
+        return jsonBody;
+    }
+
+    private JSONObject handlePost(
+            org.apache.axis2.context.MessageContext axis2MessageContext) throws UserStoreException, IOException {
+        LOG.info("Handling POST");
+        if (!JsonUtil.hasAJsonPayload(axis2MessageContext)) {
+            return Utils.createJsonErrorObject("JSON payload is missing");
+        }
+        JsonObject payload = Utils.getJsonPayload(axis2MessageContext);
+        if (payload.has(USER_ID) && payload.has(PASSWORD)) {
+            String[] roleList = null;
+            if (payload.has(IS_ADMIN)) {
+                String adminRole = getRealmConfiguration().getAdminRoleName();
+                roleList = new String[]{adminRole};
+            }
+            String user = payload.get(USER_ID).getAsString();
+            getUserStore().addUser(user, payload.get(PASSWORD).getAsString(),
+                                   roleList, null, null);
+
+            JSONObject jsonBody = new JSONObject();
+            jsonBody.put(USER_ID, user);
+            jsonBody.put(STATUS, "Added");
+            return jsonBody;
+        } else {
+            throw new IOException("Missing one or more of the fields, '" + USER_ID + "', '" + PASSWORD + "' in the "
```

Comment:
Why are we throwing an IOException here and catching it in the invoke method? Can we look into a better way?

Reply:
What's the issue with doing it this way? For this method, it is an error in the request payload hence an IO Exception, and for the invoke method, an IO Exception is a bad request. I thought of it that way. Do you see any issue with that?

Reply:
I think it is fine. I only considered this part of the method. But with a higher picture, it seems reasonable.